### PR TITLE
:book: infrastructure provider security guidelines

### PR DESCRIPTION
**Pull request by NareshKoduru**
relates to https://github.com/kubernetes-sigs/cluster-api/issues/6519#issuecomment-2069579585

/area documentation

### CODE REVIEW

File: docs/book/src/security/infrastructure-provider-security-guidance.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Infrastructure Provider Security Guidance
```
Comment (fabriziopandini) [inline docs/book/src/security/infrastructure-provider-security-guidance.md:1]: I would add a brief intro, eg.

```suggestion
# Infrastructure Provider Security Guidance

There are few critical areas that any infrastructure provider implementer should take care of in order to ensure secure operations:
- Management of cloud credentials assigned to the infrastructure provider, including setting corresponding quotas, rate limiting, etc.
- Ensure secure access to VM for troubleshooting
- Ensuring control of all the manual operations performed on the cloud infrastructure targeted by the provider
- Housekeeping of the cloud infrastructure targeted by the provider

Following list is a community maintained list of high level recommendations; everyones help is welcome and necessary to continuously improve such list.

Each provider implementer is responsible to translate those recommendations in the context of each cloud provider.


```

Comment (NareshKoduru) [reply docs/book/src/security/infrastructure-provider-security-guidance.md:1]: It is addressed in commit a512d28. Please re-review. Thanks

**Review by fabriziopandini — COMMENTED** [review]
Thanks for taking care of this work!
/lgtm

cc @sbueringer @chrischdi for a final pass

**Comment by sbueringer** [comment]
Thx!
/approve

**Pull request merged**